source/io/module_eggShell.js

```diff
@@ -320,6 +322,8 @@
         var arrayRankPointer = Module._malloc(4);
         var vireoObjectPointer = Module._malloc(4);
         var vireoTypePointer = Module._malloc(4);
+		
+		var objectRefNumToJsObjectMap = {};
```

Comment:
We should use a Map instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map